python/dgl/runtime/spmv.py

```diff
@@ -4,159 +4,126 @@
 from ..base import DGLError
 from .. import backend as F
 from .. import utils
+from .. import ndarray as nd
+from ..graph_index import from_coo
 
 from . import ir
 from .ir import var
 
-def analyze_v2v_spmv(graph, mfunc, rfunc):
-    """Analyze if SPMV from node space to node space can be applied.
+
+def gen_v2v_spmv_schedule(graph, mfunc, rfunc, src_frame, dst_frame,
+                          edge_frame, out, out_size, src_map=None,
+                          dst_map=None, edge_map=None, out_map=None):
```

Comment:
That's very correct. The reason I have them here is 1) to make code general in case we need it; 2) improve the readability of the logic in message function where lhs and rhs is chosen (https://github.com/dmlc/dgl/blob/6720483d7eae701b51fb733578a0f1e51a75a58b/python/dgl/function/message.py#L53)